app.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+import traceback
+
+from flask import Flask, jsonify, request
+
+import MySQLdb
+
+app = Flask(__name__)
+
+
+class SimpleMySQL:
+    """
+    Provides a cursor to a connection to a MySQL instance
+    in a 'with' context.
+    """
+
+    def __enter__(self):
+        """
+        Starts a connection to a MySQL instance
+        """
+        self.conn = MySQLdb.connect(
+            host=os.environ.get("MYSQL_HOST"),
+            user=os.environ.get("MYSQL_USER"),
+            port=os.environ.get("MYSQL_PORT", 3306),
+            passwd=os.environ.get("MYSQL_PASSWORD"),
+            db=os.environ.get("MYSQL_DATABASE")
+        )
+        self.cursor = self.conn.cursor()
+        return self.cursor
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        try:
+            self.conn.commit()
+        # There's a global exception listener, so we want to re-raise
+        # the exceptions.
+        # However, we need to make sure that all connections and cursors
+        # are properly closed
+        except Exception as e:
+            raise e
+        finally:
+            self.cursor.close()
+            self.conn.close()
+
+
+@app.route('/execute', methods=['post'])
+def execute():
+    req = request.json
+    query = req['query']
+    args = req.get('data', {})
+    with SimpleMySQL() as cur:
+        cur.execute(query, args)
+        return jsonify(cur.fetchall())
+
+
+class InsertBuilder:
+    """
+    Builds an insert query string for one or more insert instructions
+    """
+
+    def __init__(self):
+        self.params = []
+        self.value_strs = []
+        self.first_item = None
+
+    def add(self, values):
+        items = sorted(values.items())
+        # save the first item for the column names
+        if self.first_item is None:
+            self.first_item = items
+
+        self.params.extend(map(lambda x: x[1], items))
+        self.value_strs.append(f"({','.join(map(lambda x: '%s', values))})")
+
+    def names(self):
+        return ','.join(map(lambda x: f'{x[0]}', self.first_item))
+
+    def values(self):
+        return ','.join(self.value_strs)
+
+
+@app.route('/insert', methods=['post'])
+def insert():
+    builder = InsertBuilder()
+    req = request.json
+    table = req['table']
+    values = req['values']
+    # allow one or multiple insert entries
+    if isinstance(values, list):
+        for e in values:
+            builder.add(e)
```

Comment:
```suggestion
            map(builder.add, values)
```
(replacing the `for` loop)

A little adjustment but faster primitive in Python.

Reply:
Not necessarily (see e.g. https://stackoverflow.com/questions/43677763/why-is-in-my-case-for-loop-faster-vs-map-reduce-and-list-comprehension)

Also, it would be `lambda x: builder.add(x)` as otherwise the class context would get lost.

Anyhow, while I love functional programming I typically try to keep it to a minimum in Python as its not universally appreciated in Python.